debug-graph/src/main.rs

```diff
@@ -0,0 +1,63 @@
+use geo::HaversineLength;
+use geojson::{Feature, Geometry};
+use route_snapper_graph::RouteSnapperMap;
+
+fn main() {
+    let args: Vec<String> = std::env::args().collect();
+    if args.len() != 2 {
+        println!("Pass in a snap.bin file");
+        std::process::exit(1);
+    }
+    let bytes = std::fs::read(&args[1]).unwrap();
+    let mut map: RouteSnapperMap = bincode::deserialize(&bytes).unwrap();
+
+    // TODO Move this to route_snapper_graph
+    if !map.override_forward_costs.is_empty() && map.override_forward_costs.len() != map.edges.len()
+    {
+        panic!("override_forward_costs length doesn't match edges length",);
+    }
+    if !map.override_backward_costs.is_empty()
+        && map.override_backward_costs.len() != map.edges.len()
+    {
+        panic!("override_backward_costs length doesn't match edges length",);
+    }
+
+    for (idx, edge) in map.edges.iter_mut().enumerate() {
+        edge.length_meters = edge.geometry.haversine_length();
+
+        if map.override_forward_costs.is_empty() {
+            edge.forward_cost = Some(edge.length_meters);
+        } else {
+            edge.forward_cost = map.override_forward_costs[idx];
+        }
+
+        if map.override_backward_costs.is_empty() {
+            edge.backward_cost = Some(edge.length_meters);
+        } else {
+            edge.backward_cost = map.override_backward_costs[idx];
+        }
+    }
+
+    // This is a copy of renderGraph from the WASM API. Browsers seem to have limits for how large
```

Comment:
This entire file exists just because https://github.com/dabreegster/route_snapper/blob/7554cec5889f93e39bb9b30f42e67abddc3c1c4c/examples/index.html#L262 seems to break for large files. If you know an easy fix, I'd love to get rid of duplicate code